### PR TITLE
chore: Remove multipleTiers FF calls throughout app

### PR DESCRIPTION
**Pull request by ajay-sentry**
# Description

This PR aims to baseline and remove references to the "multipleTiers" flag throughout Gazebo. This is just a bit of a pain to work around for some tickets I have this sprint so figured I may as well just remove it and save myself the hassle later lol

Deprecation was fairly straightforward, remove all references in the non-spec files, then remove any spec that relied on the flag being "off" and update any spec which were relying on the flag to be false to show the user didn't have team plans available, and instead rename to "isTeamPlan" to represent the actual conditional logic

Aside from the code cleanup aspect, we can reduce any "funkiness" with ad-blocker affecting LD (not sure if it does, but in prior experience we used to be affected by ad blockers when using Optimizely), and reduced API calls to LD that we don't need, hopefully saving a teeny bit of cash in the process.

[FF Link
](https://app.launchdarkly.com/dashboard/production/features/multiple-tiers/targeting)
App still works after some basic QA locally.

Closes https://github.com/codecov/engineering-team/issues/2456

# Screenshots

# Link to Sample Entry

# Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: src/pages/CommitDetailPage/CommitCoverage/UploadsCard/useUploads/useUploads.js [hunk 1]
```diff
@@ -3,20 +3,15 @@ import { useParams } from 'react-router-dom'
...
 
 export function useUploads() {
   const { provider, owner, repo, commit } = useParams()
   const { data: overview } = useRepoOverview({ provider, owner, repo })
-  const { multipleTiers } = useFlags({
-    multipleTiers: false,
-  })
   const { data: tierData } = useTier({ provider, owner })
 
   // TODO: We need backend functionality to properly manage access to carryforward flags for team tier members
```
Comment (suejung-sentry) [inline src/pages/CommitDetailPage/CommitCoverage/UploadsCard/useUploads/useUploads.js:13]: Is this TODO now obsolete? TBH I have no idea what it's trying to say :joy:

File: src/pages/CommitDetailPage/CommitCoverage/UploadsCard/useUploads/useUploads.js [hunk 2]
```diff
@@ -3,20 +3,15 @@ import { useParams } from 'react-router-dom'
...
-  const isTeamPlan =
-    multipleTiers && tierData === TierNames.TEAM && overview?.private
+  const isTeamPlan = tierData === TierNames.TEAM && overview?.private
```
Comment (suejung-sentry) [inline src/pages/CommitDetailPage/CommitCoverage/UploadsCard/useUploads/useUploads.js:14]: I know your PR is not introducing this concept but I found it confusing that many files here don't separate out like `isTeamPlan` && `isRepoPrivate` and instead put them both into one arguably-misnamed variable. 

Unless I'm misunderstanding what a "teamPlan" means...

**Review by suejung-sentry — APPROVED** [review]
lgtm - thanks for doing the cleanup!

File: src/pages/CommitDetailPage/CommitCoverage/UploadsCard/useUploads/useUploads.js (hunk 1 above)
Comment (ajay-sentry) [reply src/pages/CommitDetailPage/CommitCoverage/UploadsCard/useUploads/useUploads.js:13]: im right there with ya loool, that's a TODO to figure out later 😂

File: src/pages/CommitDetailPage/CommitCoverage/UploadsCard/useUploads/useUploads.js (hunk 2 above)
Comment (ajay-sentry) [reply src/pages/CommitDetailPage/CommitCoverage/UploadsCard/useUploads/useUploads.js:14]: this is definitely something worth bringing up on slack, its a bit of a black box to me as well

**Pull request merged**